sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPatchHelper.scala

```diff
@@ -162,4 +165,113 @@ class CosmosPatchHelper(diagnosticsConfig: DiagnosticsConfig,
    throw new IllegalArgumentException(s"Increment operation is not supported for non-numeric type ${jsonNode.getClass}")
   }
  }
+
+  def createCosmosPatchBulkUpdateOperations(objectNode: ObjectNode): List[PatchOperationCore[JsonNode]] = {
+
+      val cosmosPatchBulkUpdateOperations = new ListBuffer[PatchOperationCore[JsonNode]]()
+      val fieldIterator = objectNode.fields()
+      while (fieldIterator.hasNext) {
+          val fieldEntry = fieldIterator.next()
+
+          // check whether there is customized mapping path
+          cosmosPatchConfigs.columnConfigsMap.get(fieldEntry.getKey) match {
+              case Some(userDefinedColumnConfig) =>
+                  if (!systemProperties.contains(userDefinedColumnConfig.mappingPath.substring(1))) {
+                      val node = fieldEntry.getValue
+                      val effectiveNode = if (userDefinedColumnConfig.isRawJson) {
+                          objectMapper.readTree(node.asText())
+                      } else {
+                          node
+                      }
+
+                      userDefinedColumnConfig.operationType match {
+                          case CosmosPatchOperationTypes.Set =>
+                              cosmosPatchBulkUpdateOperations +=
+                                  new PatchOperationCore[JsonNode](PatchOperationType.SET, userDefinedColumnConfig.mappingPath, effectiveNode)
+                          case _ => throw new RuntimeException(s"Patch operation type is not supported for itemPatchBulkUpdate write strategy")
+                      }
+                  }
+
+              case None =>
+                  if (!systemProperties.contains(fieldEntry.getKey)) {
+                      cosmosPatchBulkUpdateOperations +=
+                          new PatchOperationCore[JsonNode](PatchOperationType.SET, s"/${fieldEntry.getKey}", fieldEntry.getValue)
+                  }
+          }
+      }
+
+      cosmosPatchBulkUpdateOperations.toList
+  }
+
+  def patchBulkUpdateItem(
+                             itemToBeUpdatedOpt: Option[ObjectNode],
+                             patchBulkUpdateOperations: List[PatchOperationCore[JsonNode]]): ObjectNode = {
+
+      // Do not do the modification on original objectNode, as it maybe used by other patchBulkUpdate operations
+      val rootNode = itemToBeUpdatedOpt match {
+          case Some(itemToBeUpdate) => Utils.getSimpleObjectMapper.createObjectNode().setAll(itemToBeUpdate)
+          case _ => Utils.getSimpleObjectMapper.createObjectNode()
+      }
+
+      patchBulkUpdateOperations.foreach(patchOperation => {
+          patchBulkUpdateFields(rootNode, patchOperation.getOperationType, patchOperation.getPath, patchOperation.getResource)
+      })
+
+      rootNode
+  }
+
+  private[this] def patchBulkUpdateFields(
```

Comment:
NIT - this method and the one below look like it would be good to have explicit unit testing for them. I would consider making them package private and add explicit unit tests just for each of these two methods

Reply:
still keep this method object private level, but added tests for `patchBulkUpdateItem` which will internally call these two methods